metrics.go

```diff
@@ -58,14 +58,20 @@ func (m MetricType) validateValue(value string) error {
 
 // Metric represents a single metric definition
 type Metric struct {
-	Type        MetricType
-	Description string
+	Type        MetricType `yaml:"type"`
+	Description string     `yaml:"description"`
+}
+
+// Plan represents the plan section of metrics.yaml
+type Plan struct {
+	Required bool `yaml:"required"`
 }
 
 // Metrics contains the metrics declarations encoded in the metrics.yaml
 // file.
 type Metrics struct {
-	Metrics map[string]Metric
+	Metrics map[string]Metric `yaml:"metrics"`
+	Plan    *Plan             `yaml:"plan,omitempty"`
```

Comment:
If this was a value rather than a pointer, then you wouldn't need to check nil below -- it'd just have the default value when omitted.